test/e2e/elasticache/tests/test_replicationgroup.py

```diff
@@ -80,6 +80,25 @@ def rg_input_coverage(bootstrap_resources, make_rg_name, make_replication_group,
     sleep(DEFAULT_WAIT_SECS)
     rg_deletion_waiter.wait(ReplicationGroupId=input_dict["RG_ID"]) #throws exception if wait fails
 
+@pytest.fixture(scope="module")
+def rg_auth_token(make_rg_name, make_replication_group, rg_deletion_waiter):
+    k8s.create_k8_secret("default", "first", "secret1", "securetoken123456")
+    input_dict = {
+        "RG_ID": make_rg_name("rg-auth-token"),
+        "NAME": "first",
+        "KEY": "secret1"
+    }
+    k8s.create_k8_secret("default", "second", "secret2", "newsecuretoken123456")
```

Comment:
Instead of creating multiple secrets and a replication group in one fixture, separate out the secret management into its own fixture and have this fixture depend that secret-providing fixture.

See https://docs.pytest.org/en/latest/how-to/fixtures.html#safe-teardowns for more details about this.

```python
@pytest.fixture(scope="module")
def first_secret():
    k8s.create_k8_secret("default", "first", "secret1", "securetoken123456")
    yield
    k8s.delete_secret("default", "first")

@pytest.fixture(scope="module")
def second_secret():
    k8s.create_k8_secret("default", "second", "secret2", "newsecuretoken123456")
    yield
    k8s.delete_secret("default", "second")

pytest.fixture(scope="module")
def rg_auth_token(make_rg_name, make_replication_group, rg_deletion_waiter, first_secret, second_secret):
    input_dict = {
        "RG_ID": make_rg_name("rg-auth-token"),
        "NAME": "first",
        "KEY": "secret1"
    }
    (reference, resource) = make_replication_group("replicationgroup_authtoken", input_dict, input_dict["RG_ID"])
    yield (reference, resource)
    k8s.delete_custom_resource(reference)
    sleep(DEFAULT_WAIT_SECS)
    rg_deletion_waiter.wait(ReplicationGroupId=input_dict["RG_ID"]) #throws exception if wait fails
```

Reply:
Fixed, Thank you!